### PR TITLE
Tk4/0009 kabi 

### PR DESCRIPTION
**Pull request by arafatms**
1.Some optimization for unixbench & iozone;
2.Bugfix backported from tkernel

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/TencentOS-kernel?pullRequest=307) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/TencentOS-kernel?pullRequest=307) before we can accept your contribution.<br/>**2** out of **14** committers have signed the CLA.<br/><br/>:white_check_mark: frankjpliu<br/>:white_check_mark: xmmgithub<br/>:x: johnnyaiai<br/>:x: axboe<br/>:x: torvalds<br/>:x: sargun<br/>:x: zhangyi089<br/>:x: alan-maguire<br/>:x: yang-shi<br/>:x: hygoni<br/>:x: goldwynr<br/>:x: herberthbli<br/>:x: brookxu-tx<br/>:x: ultra-nexus<br/><hr/>**johnnyaiai** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/TencentOS-kernel?pullRequest=307) it.</sub>

**Comment by flying-122** [comment]
config: Enable CONFIG_PREEMPT_NONE by default
--这个设置影响较大，CONFIG_PREEMPT_NONE是内核不能抢占，可能在多数情况下都会影响业务的响应延时。

File: drivers/android/binder.c [hunk 1]
```diff
@@ -2232,10 +2232,12 @@ static void binder_deferred_fd_close(int fd)
 		return;
```
Comment (sclaibin) [inline drivers/android/binder.c:2232]: 这种backport 在第一行加上 upstream commit: xxxx  ，然后在最后一行加上自己的Signed-off-by.
这样一是方便review时查看上游补丁，二是通过Signed-off-by找到具体bakcport的人，方便进行后续回溯

File: fs/file.c [hunk 2]
```diff
@@ -725,6 +725,10 @@ static struct file *__fget(unsigned int fd, fmode_t mask, unsigned int refs)
 			file = NULL;
```
Comment (sclaibin) [inline fs/file.c]: 在commit里可以加上自己的signed off by

File: fs/file.c [hunk 3]
```diff
@@ -708,9 +708,9 @@ void do_close_on_exec(struct files_struct *files)
 	spin_unlock(&files->file_lock);
```
Comment (sclaibin) [inline fs/file.c:708]: 加上upstream commit and signed off by

File: package/default/config.default [hunk 4]
```diff
@@ -90,8 +90,8 @@ CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
 # end of Timers subsystem
 
-# CONFIG_PREEMPT_NONE is not set
-CONFIG_PREEMPT_VOLUNTARY=y
```
Comment (sclaibin) [inline package/default/config.default:94]: 调度抢占参数的变更会影响较大，这里是否要改为完全不抢占 需要再讨论下

File: package/default/config.default [hunk 5]
```diff
@@ -86,6 +86,8 @@ CONFIG_NO_HZ_COMMON=y
 # CONFIG_HZ_PERIODIC is not set
```
Comment (sclaibin) [inline package/default/config.default:86]: 建议commit有个主题，描述下CONFIG打开或者关闭解决的问题

File: package/default/config.default [hunk 6]
```diff
@@ -4997,7 +4997,7 @@ CONFIG_RCU_CPU_STALL_TIMEOUT=60
 # CONFIG_NOTIFIER_ERROR_INJECTION is not set
 CONFIG_FUNCTION_ERROR_INJECTION=y
 # CONFIG_FAULT_INJECTION is not set
-CONFIG_LATENCYTOP=y
+# CONFIG_LATENCYTOP is not set
```
Comment (sclaibin) [inline package/default/config.default:5001]: latencytop_enabled 默认为关闭，对运行时性能是没有影响的，这里直接从编译里去掉的目的是什么（是实测发现有性能影响吗）？

File: fs/ext4/inode.c [hunk 7]
```diff
@@ -3051,35 +3051,37 @@ static int ext4_da_write_end(struct file *file,
 	end = start + copied - 1;
```
Comment (sclaibin) [inline fs/ext4/inode.c:2993]: 可以考虑加下自己的signed by

File: tools/bpf/bpftool/net.c [hunk 8]
```diff
@@ -11,7 +11,6 @@
 #include <bpf/bpf.h>
```
Comment (sclaibin) [inline tools/bpf/bpftool/net.c:11]: commit加上upstream commit and signed by

File: package/default/config.default [hunk 9]
```diff
@@ -145,7 +145,7 @@ CONFIG_ARCH_SUPPORTS_NUMA_BALANCING=y
 CONFIG_ARCH_WANT_BATCHED_UNMAP_TLB_FLUSH=y
 CONFIG_ARCH_SUPPORTS_INT128=y
 CONFIG_NUMA_BALANCING=y
-# CONFIG_NUMA_BALANCING_DEFAULT_ENABLED is not set
+CONFIG_NUMA_BALANCING_DEFAULT_ENABLED=y
```
Comment (sclaibin) [inline package/default/config.default:148]: 根据以前的经验，NUMA BALANCE 在很多场景里都有负优化的表现，不建议改为default enable

File: include/linux/pagemap.h [hunk 10]
```diff
@@ -455,6 +455,43 @@ static inline pgoff_t linear_page_index(struct vm_area_struct *vma,
 	return pgoff;
 }
```
Comment (sclaibin) [inline include/linux/pagemap.h:456]: 可以加下signed by

File: include/linux/fs.h [hunk 11]
```diff
@@ -316,6 +316,8 @@ enum rw_hint {
 #define IOCB_SYNC		(1 << 5)
 #define IOCB_WRITE		(1 << 6)
```
Comment (sclaibin) [inline include/linux/fs.h:317]: 加下upstream commit id

File: package/default/config.default (hunk 6 above)
Comment (arafatms) [reply package/default/config.default:5001]: latencytop在copy_process的时候调用clear_tsk_latency_tracing（不管latencytop_enabled是否打开）这个函数里多核竞争一个锁，会导致性能下降1-2%

**Pull request closed without merging**